### PR TITLE
mmds: deprecate V1

### PR DESCRIPTION
**Pull request by alindima**
## Description of Changes

Deprecates MmdsV1

- [ ] This functionality can be added in [`rust-vmm`][1].

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The issue which led to this PR has a clear conclusion.
- [x] This PR follows the solution outlined in the related issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes follow the [Runbook for Firecracker API changes][2].
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

[1]: https://github.com/rust-vmm
[2]: ../docs/api-change-runbook.md

### CODE REVIEW

File: tests/integration_tests/functional/test_mmds.py [hunk 1]
```diff
@@ -1003,3 +1005,42 @@ def test_mmds_v2_negative(test_microvm_with_api, network_config):
...
+    assert 'deprecation' not in response.headers
+
+    test_microvm.start()
+    lines = metrics_fifo.sequential_reader(100)
+
+    assert sum(list(map(
+        lambda line:
+            json.loads(line)['deprecated_api']['deprecated_http_api_calls'],
+        lines
+    ))) == 2
```
Comment (luminitavoicu) [inline tests/integration_tests/functional/test_mmds.py:1046]: Nice!

**Review by luminitavoicu — APPROVED** [review]
Looks good!

**Comment by raduiliescu** [comment]
Do we want to release this in Firecracker v1.1?

**Review by georgepisaltu — COMMENTED** [review]
Maybe it's worth considering changing the default MMDS version to V2 so that you don't default to a deprecated endpoint.

**Pull request merged**